v22.2/backup-and-restore-monitoring.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Backup and Restore Monitoring
+summary: An overview of the metrics to monitor backup and restore jobs in CockroachdB.
+toc: true
+docs_area: manage
+---
+
+CockroachDB includes metrics to monitor [backup](backup.html), [restore](restore.html), and [scheduled backup](create-schedule-for-backup.html) jobs. You can use monitoring integrations to alert when there are anomalies, such as backups that have failed or restore jobs encountering a retryable error.
+
+Depending on whether you are using a {{ site.data.products.dedicated }} or {{ site.data.products.core }} cluster, you can use the following to monitor backup and restore metrics for your CockroachDB cluster:
+
+- [Prometheus endpoint](#prometheus-endpoint): {{ site.data.products.dedicated }}, {{ site.data.products.core }}
+- [Datadog integration](#datadog-integration): {{ site.data.products.dedicated }}
+
+We recommend setting up monitoring to alert when anomalies occur. You can then use the following SQL statements to inspect details relating to schedules, jobs, and backups:
+
+- [`SHOW SCHEDULES`](show-schedules.html)
+- [`SHOW JOBS`](show-jobs.html)
+- [`SHOW BACKUP`](show-backup.html)
+
+For detail on [managed-service backups](../cockroachcloud/use-managed-service-backups.html) that Cockroach Labs stores for your {{ site.data.products.db }} cluster, see the **Backups** page in the Cloud Console.
+
+{% include {{ page.version.version }}/backups/metrics-per-node.md %}
+
+## Prometheus endpoint
+
+You can access the [Prometheus endpoint](monitoring-and-alerting.html#prometheus-endpoint) (`http://<host>:<http-port>/_status/vars`) for backup and restore metrics with **{{ site.data.products.dedicated }} or {{ site.data.products.core }}** clusters.
+
+See the [Monitor CockroachDB with Prometheus](monitor-cockroachdb-with-prometheus.html) tutorial for guidance on installing and setting up Prometheus and Alertmanager to track metrics.
+
+### Available metrics
+
+We recommend the following guidelines:
+
+- Use the `schedules_backup_last_completed_time` metric. Configure this metric to monitor the backup schedule you would use in a disaster.
+- Monitor for a lack of successful backups rather than failed backups. A stuck scheduled job will not increment the `schedules_backup_failed` metric. Instead, alert on the `schedules_backup_last_completed_time` metric if the timestamp has not moved forward as expected.
+
+Metric | Description 
+-------+-------------
```

Comment:
This metrics section has multiple problems, few of which can be fixed all the way downstream here on the Docs team:

Metrics use dot notation to demarcate along metrics families, as seen [here](https://www.cockroachlabs.com/docs/stable/metrics.html) and in http://localhost:8080/_admin/v1/chartcatalog.

For example:
- `schedules_round_reschedule_skip` should be `schedules.round.reschedule-skip`
- `schedules_backup_last_completed_time` should be `schedules.BACKUP.last-completed-time`
- etc.

There is no rubric by which underscore characters may be intuited to their intended character replacement, so you'll have to go through each by hand via guess and check. I recommend http://localhost:8080/_admin/v1/chartcatalog and trying single words at a time, since a `_` could be any of `.` `-` or `_`.

Did engineering give you a resource like this one: https://github.com/cockroachlabs/managed-service/blob/master/pkg/otel/assets/metrics.yaml

If so, that uses some strange internal representation that does not match what is shown to end users in the third-party integrations named on this page.

Further, the descriptions here, while helpful, are manual and do not match the metrics descriptions shown to users in the third-party integrations named on this page, in the DB Console custom chart: http://localhost:8080/#/debug/chart (click `Select a metric ...`) and in our own: http://localhost:8080/_admin/v1/chartcatalog

I don't know what to do about this, I haven't solved this on my end either in Obs space.

@dt can you confirm the above please.

Big picture: This begins a slippery slope, where we list metrics in multiple places, where they can now get out of date at twice the volume. I don't know what to do about this either.

Reply:
@andf-crl I picked them up "literally" from the status/vars endpoint hence the underscores.

- Re the descriptions, I think we wanted to be able to add detail to the metrics here (as a docs page) where we felt necessary for guidance. This feels akin to most other areas of the docs site? Is the metrics page you link to automatically pulled then? My sense is we want to move forward with this page and add nuanced guidance where needed. Let me know if I haven't captured your meaning here.

Reply:
Ah gotcha, thank you. My understanding is that `_status/vars` is more of a consumable API rather than a notation with end users in mind. However, it is indeed what Prometheus consumes to populate its output, so it is indeed the notation that will be shown to Prometheus users in Prometheus, you're right. I withdraw that comment, then, given that this doc is essentially all about metrics surfaced through Prometheus, and the different notation that appears in the other three locations is essentially out of scope to this particular doc.

Re: descriptions: the intent is indeed to automatically populate our currently-out-of-date metrics docs reference, but I have not been able to solve the fundamental problems raised above, so am unlikely to be able to enact this in the short term. Since the short term is all any of us have control over: I withdraw my comment on descriptions as well.

Apologies: I won't get any further in the way of this PR.

Reply:
@andf-crl No, no! Definitely not getting in the way, I appreciate your expertise here! To your comment, I will keep as is for now. On the second point, it would be good to stay in the loop with your intentions on the overarching metrics docs, I would generally rather that we can stay consistent across the docs. If there is a way to also add some nuance on the page, while still automating I would be into that. This is most definitely a new page, and therefore, it's very earliest iteration! As anything comes up around the metrics work you're doing, please feel free to open issues for me to update this page, or directly alter the page!